packages/fields/src/components/create-template-part-modal/index.tsx

```diff
@@ -16,59 +15,107 @@ import {
 	__experimentalHStack as HStack,
 	__experimentalVStack as VStack,
 } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
-import { useState } from '@wordpress/element';
 import { useInstanceId } from '@wordpress/compose';
-import { store as noticesStore } from '@wordpress/notices';
 import { store as coreStore } from '@wordpress/core-data';
-import { check } from '@wordpress/icons';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	check,
+	footer as footerIcon,
+	header as headerIcon,
+	sidebar as sidebarIcon,
+	symbolFilled as symbolFilledIcon,
+} from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+// @ts-ignore
 import { serialize } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import {
-	TEMPLATE_PART_POST_TYPE,
-	TEMPLATE_PART_AREA_DEFAULT_CATEGORY,
-} from '../../store/constants';
-import {
-	useExistingTemplateParts,
-	getUniqueTemplatePartTitle,
 	getCleanTemplatePartSlug,
+	getUniqueTemplatePartTitle,
+	useExistingTemplateParts,
 } from './utils';
 
+type CreateTemplatePartModalContentsProps = {
+	defaultArea?: string;
+	blocks: any[];
+	confirmLabel?: string;
+	closeModal: () => void;
+	onCreate: ( templatePart: any ) => void;
+	onError?: () => void;
+	defaultTitle?: string;
+};
+
+/**
+ * A React component that renders a modal for creating a template part. The modal displays a title and the contents for creating the template part.
+ * This component should not live in this package, it should be moved to a dedicated package responsible for managing template.
+ * @param {Object} props            The component props.
+ * @param          props.modalTitle
+ */
 export default function CreateTemplatePartModal( {
 	modalTitle,
 	...restProps
-} ) {
+}: {
+	modalTitle: string;
+} & CreateTemplatePartModalContentsProps ) {
 	const defaultModalTitle = useSelect(
 		( select ) =>
-			select( coreStore ).getPostType( TEMPLATE_PART_POST_TYPE )?.labels
+			// @ts-ignore
```

Comment:
Is this `// @ts-ignore` necessary?

In general, I usually recommend using `@ts-expect-error` instead of `@ts-ignore` ([this article](https://dev.to/maafaishal/ts-expect-error-over-ts-ignore-in-typescript-5140) explains the reason). And in general, it's always a good idea to leave an inline comment explain the reason why we're ignoring / expecting a given TS error.

Reply:
Nice catch! I overlooked it when I iterate on this PR. Fixed with https://github.com/WordPress/gutenberg/pull/67709